src/common/cockpitchannel.c

```diff
@@ -351,14 +351,20 @@ cockpit_channel_actual_send (CockpitChannel *self,
   /* A wraparound of our gint64 size? */
   if (priv->flow_control)
     {
+      gboolean under_pressure;
       size = g_bytes_get_size (payload);
       g_return_if_fail (G_MAXINT64 - size > priv->out_sequence);
 
       /* How many bytes have been sent (queued) */
       out_sequence = priv->out_sequence + size;
 
-      /* Every CHANNEL_FLOW_PING bytes we send a ping */
-      if (out_sequence / CHANNEL_FLOW_PING != priv->out_sequence / CHANNEL_FLOW_PING)
+      /* If we've sent more than the window, we just got under pressure;
+       * do an edge trigger instead of level trigger to avoid ping/signal loops */
+      under_pressure = (priv->out_sequence <= priv->out_window) && (out_sequence > priv->out_window);
```

Comment:
If this is edge triggered then the variable name needs to change. The variable is named for a level.

Reply:
Fair enough. I renamed it to `trigger_pressure`.